dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/event/MetricsEventBus.java

```diff
@@ -63,27 +68,49 @@ public static <T> T post(MetricsEvent event, Supplier<T> targetSupplier) {
      */
     public static <T> T post(MetricsEvent event, Supplier<T> targetSupplier, Function<T, Boolean> trFunction) {
         T result;
-        before(event);
+        tryInvoke(() -> {
+            before(event);
+        });
         if (trFunction == null) {
             try {
                 result = targetSupplier.get();
             } catch (Throwable e) {
-                error(event);
+                tryInvoke(() -> {
+                    error(event);
+                });
                 throw e;
             }
-            after(event, result);
+            tryInvoke(() -> {
+                after(event, result);
+            });
         } else {
             // Custom failure status
             result = targetSupplier.get();
             if (trFunction.apply(result)) {
-                after(event, result);
+                tryInvoke(() -> {
+                    after(event, result);
+                });
             } else {
-                error(event);
+                tryInvoke(() -> {
+                    error(event);
+                });
             }
         }
         return result;
     }
 
+    interface TryCallBack {
+         void callback();
+    }
+
+    public static void tryInvoke(TryCallBack tryCallBack) {
+        try {
+            tryCallBack.callback();
+        } catch (Throwable e2) {
+            logger.debug("MetricsEventBus tryInvoke error", e2);
```

Comment:
```suggestion
            logger.warn("MetricsEventBus tryInvoke error", e2);
```

warn with error code